lib/aws/rails/railtie.rb

```diff
@@ -46,65 +36,37 @@ class Railtie < ::Rails::Railtie
           config.eager_load_namespaces << Aws
         end
       end
-    end
-
-    # Configures the AWS SDK for Ruby's logger to use the Rails logger.
-    def self.log_to_rails_logger
-      Aws.config[:logger] = ::Rails.logger
-      nil
-    end
 
-    # Configures the AWS SDK with credentials from Rails encrypted credentials.
-    def self.use_rails_encrypted_credentials
-      # limit the config keys we merge to credentials only
-      aws_credential_keys = %i[access_key_id secret_access_key session_token account_id]
-      creds = ::Rails.application.credentials[:aws].to_h.slice(*aws_credential_keys)
-      Aws.config.merge!(creds)
-    end
-
-    # This is called automatically from the SDK's Railtie, but can be manually
-    # called if you want to specify options for building the Aws::SES::Client or
-    # Aws::SESV2::Client.
-    #
-    # @param [Symbol] name The name of the ActionMailer delivery method to
-    #   register, either :ses or :sesv2.
-    # @param [Hash] client_options The options you wish to pass on to the
-    #   Aws::SES[V2]::Client initialization method.
-    def self.add_action_mailer_delivery_method(name = :ses, client_options = {})
-      # TODO: remove this method in aws-sdk-rails ~> 5
-      ActiveSupport.on_load(:action_mailer) do
-        if name == :sesv2
-          add_delivery_method(name, Aws::Rails::Sesv2Mailer, client_options)
-        else
-          add_delivery_method(name, Aws::Rails::SesMailer, client_options)
+      # Add ActiveSupport Notifications instrumentation to AWS SDK client operations.
+      # Each operation will produce an event with a name `<operation>.<service>.aws`.
+      # For example, S3's put_object has an event name of: put_object.S3.aws
+      initializer 'aws-sdk-rails.instrument-sdk-operations', after: :load_active_support do
```

Comment:
I'm not sure this should be on by default - I think theres a cost of the extra plugin + processing the notifications.  Additionally, might be order sensitive - this will add plugins only to services that have already been required

Reply:
I debated this - load_active_support is very early in the chain, executed 2nd https://guides.rubyonrails.org/configuring.html#initializers. I think even if you add this thing as a custom initializer (which we recommended already) it is effectively being done the same way, while the application is initializing. It would have already required users to require the sdk before doing so.

I think the performance impact is fairly small. On the ActiveSupport side, it's a block being called by wrapper, and on the plugin side, it's just an additional plugin which we add frequently.

So in short, I'm not sure what to do about it. We could swing the other way and have nothing with a railtie, but I think it's nice to have these things automatically registered.

Reply:
yeah, if you think its low cost, then that seems fine to keep by default.